src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -951,6 +997,51 @@ protected function isJsonCastable($key)
         return $this->hasCast($key, ['array', 'json', 'object', 'collection']);
     }
 
+    /**
+     * Getting the execution result from a user Cast object.
+     *
+     * @param  string  $key
+     * @param  mixed  $value
+     * @return mixed
+     */
+    protected function fromCustomCastable($key, $value = null)
+    {
+        return $this
+            ->normalizeCastToCallable($key)
+            ->fromDatabase($key, $value);
+    }
+
+    /**
+     * Converting a value by custom Cast.
+     *
+     * @param  string  $key
+     * @param  mixed  $value
+     * @return mixed
+     */
+    protected function toCustomCastable($key, $value = null)
+    {
+        return $this
+            ->normalizeCastToCallable($key)
+            ->toDatabase($key, $value);
+    }
+
+    /**
+     * Getting a custom cast instance.
+     *
+     * @param  string  $key
+     * @return \Illuminate\Contracts\Database\Eloquent\Castable
+     */
+    protected function normalizeCastToCallable($key)
+    {
+        if (! isset(static::$castsCache[$key])) {
+            $cast = $this->getCast($key);
+
+            static::$castsCache[$key] = new $cast;
```

Comment:
```suggestion
            static::$castsCache[$key] = Container::getInstance()->make($cast);
```
… and import the container.

Then we've the best of all suggested options ( https://github.com/laravel/framework/pull/30958#discussion_r361830332 )

Reply:
Now I understand. Thanks! 🙏

Done.

PS: how hard it is to communicate through a translator 😀

Reply:
> PS: how hard it is to communicate through a translator 😀

Sorry to hear that but I think you're doing great 😄

Reply:
> Sorry to hear that but I think you're doing great 😄

Thanks! 🍻